### PR TITLE
master

### PR DESCRIPTION
**Pull request by rensaketh**
Added algorithm implementing length of longest substring with no repeating characters

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1893"></a>

### CODE REVIEW

File: strings/CMakeLists.txt [hunk 1]
```diff
@@ -7,7 +7,7 @@ file( GLOB APP_SOURCES RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} *.cpp )
 foreach( testsourcefile ${APP_SOURCES} )
     # I used a simple string replace, to cut off .cpp.
     string( REPLACE ".cpp" "" testname ${testsourcefile} )
-    add_executable( ${testname} ${testsourcefile} )
+    add_executable( ${testname} ${testsourcefile} longest_substr_no_repeating.cpp)
```
Comment (Panquesito7) [inline strings/CMakeLists.txt:10]: Please revert the changes in this file. Thanks.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Great work! 🎉
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/rensaketh/C-Plus-Plus/actions

File: strings/CMakeLists.txt (hunk 1 above)
Comment (rensaketh) [reply strings/CMakeLists.txt]: Just reverted the change

File: strings/CMakeLists.txt [hunk 2]
```diff
@@ -7,7 +7,7 @@ file( GLOB APP_SOURCES RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} *.cpp )
 foreach( testsourcefile ${APP_SOURCES} )
     # I used a simple string replace, to cut off .cpp.
     string( REPLACE ".cpp" "" testname ${testsourcefile} )
-    add_executable( ${testname} ${testsourcefile} )
+    add_executable( ${testname} ${testsourcefile})
```
Comment (Panquesito7) [inline strings/CMakeLists.txt:10]:
```suggestion
    add_executable( ${testname} ${testsourcefile} )
```

File: strings/longest_substr_no_repeating.cpp [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    * Given a string we are required to find the
+    * longest possible substring without any repeating characters.
+    * The time and space complexity are both O(n) since an unordered set is used
+    * and the entire string is traversed.
+* @author [Saketh Renangi](https://github.com/rensaketh)
+*/
+
+#include <cassert>   /// for assert
+#include <unordered_set>  /// for unordered_set
+#include <string> //for string
```
Comment (Panquesito7) [inline strings/longest_substr_no_repeating.cpp:14]:
```suggestion
#include <cassert>         /// for assert
#include <unordered_set>  /// for unordered_set
#include <string>        /// for string
```

File: strings/longest_substr_no_repeating.cpp [hunk 4]
```diff
@@ -0,0 +1,67 @@
+/*
+* @file
+* @brief Sliding window solution to find the longest substring with no repeating characters
+* @details
+    * Given a string we are required to find the
+    * longest possible substring without any repeating characters.
+    * The time and space complexity are both O(n) since an unordered set is used
+    * and the entire string is traversed.
```
Comment (Panquesito7) [inline strings/longest_substr_no_repeating.cpp:8]:
```suggestion
 * Given a string we are required to find the
 * longest possible substring without any repeating characters.
 * The time and space complexity are both O(n) since an unordered set is used
 * and the entire string is traversed.
```

File: strings/longest_substr_no_repeating.cpp [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+ * @brief This function uses the sliding window approach to
+ * find the longest substring that does not have a repeating character
+ *
+ * @param String named s, used to find substrings from this string
+ *
+ * @returns Retuns an integer that is the length of the longest substring
+ * with no repeated characters
+ */
+
+int lengthOfLongestSubstring(std::string s) {
```
Comment (Panquesito7) [inline strings/longest_substr_no_repeating.cpp:26]: Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let us know if you need any help here or in our [Discord server](https://discord.gg/c7MnfGFGa6).** 🙂

File: strings/longest_substr_no_repeating.cpp [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+ * @brief Main function
+ * @param argc commandline argument count (ignored)
+ * @param argv commandline array of arguments (ignored)
+ * @returns 0 on exit
+ */
+int main(int argc, char *argv[]) {
+    test();  // run self-test implementations
+    // code here
+    return 0;
+}
```
Comment (Panquesito7) [inline strings/longest_substr_no_repeating.cpp:67]: Missing an endline. Without this endline, there could be issues with some IDEs and configurations.
```suggestion
}

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Almost there! 😄

**Comment by rensaketh** [comment]
I finished implementing all the changes
